app/src/main/kotlin/org/open/smsforwarder/ui/feedback/FeedbackState.kt

```diff
@@ -0,0 +1,14 @@
+package org.open.smsforwarder.ui.feedback
+
+import org.open.smsforwarder.utils.Resources
+
+data class FeedbackState(
+    val emailInputError: Resources.StringProvider? = null,
+    val bodyInputError: Resources.StringProvider? = null,
+    val emailInput: String? = null,
+    val bodyInput: String? = null
+) {
+    val submitButtonEnabled =
+        emailInputError == null && bodyInputError == null
+                && !emailInput.isNullOrBlank() && !bodyInput.isNullOrBlank()
```

Comment:
it's hard to read and understand. Split this into 2 private functions: for example 'hasInputErrors' and 'hasValues'